pkg/goDB/conditions/node/filter_direction.go

```diff
@@ -0,0 +1,160 @@
+package node
+
+import (
+	"errors"
+	"fmt"
+	"github.com/els0r/goProbe/pkg/types"
+	"github.com/els0r/goProbe/pkg/types/hashmap"
+)
+
+// FilterTypeDirection denotes filters wrt. directionality of traffic
+type FilterTypeDirection string
+
+const (
+	filterTypeDirectionIn         FilterTypeDirection = "in"
+	filterTypeDirectionInSugared                      = "inbound"
+	filterTypeDirectionOut                            = "out"
+	filterTypeDirectionOutSugared                     = "outbound"
+	filterTypeDirectionUni                            = "uni"
+	filterTypeDirectionUniSugared                     = "unidirectional"
+	filterTypeDirectionBi                             = "bi"
+	filterTypeDirectionBiSugared                      = "bidirectional"
+)
+
+var (
+	// errMultipleDirectionFilterConditions indicates that
+	// the specified condition contains too many traffic direction
+	// filters
+	errMultipleDirectionFilterConditions = errors.New("multiple direction filters specified")
+
+	// errMisplacedDirectionFilterCondition indicates that
+	// the specified condition contains a misplaced traffic direction
+	// filter
+	errMisplacedDirectionFilterCondition = errors.New("misplaced direction filter")
+
+	unsupportedDirectionFilterComparatorStr = "unsupported direction filter comparator: %s"
+	unsupportedDirectionFilterStr           = "unsupported direction filter: %s (use one of 'in', 'out', 'uni', 'bi')"
+)
+
+// isDirectionCondition returns an error if node represents a
+// direction filter condition, and the node itself otherwise.
+// It is used in combination with node.transform() in order
+// to ensure direction filter conditions occur only at the
+// top level of the node or the second level in case the
+// top level represents a conjunction (AND).
+func isDirectionCondition(node conditionNode) (Node, error) {
+	if node.attribute == types.FilterKeywordDirection {
+		return nil, errMisplacedDirectionFilterCondition
+	}
+	return node, nil
+}
+
+// extractDirectionFilter returns a ValFilter if node represents a valid direction filter condition.
+// If node represents an invalid direction filter (e.g., unsupported direction filter comparator),
+// nil is returned with an error describing the invalidity reason.
+// If the node does not contain a filter condition, nil is returned without any error.
+func extractDirectionFilter(node Node) (hashmap.ValFilter, error) {
+	switch node := node.(type) {
+	case conditionNode:
```

Comment:
Nitpick: Since you only have two possible `switch` cases, avoid the nested switch and use a simple `if` with a default `return nil, nil` after the block, increases readability IMHO.